src/polyline.js

```diff
@@ -19,17 +19,16 @@ function py2_round(value) {
 function encode(current, previous, factor) {
     current = py2_round(current * factor);
     previous = py2_round(previous * factor);
-    var coordinate = current - previous;
-    coordinate <<= 1;
-    if (current - previous < 0) {
-        coordinate = ~coordinate;
+    var coordinate = (current - previous) * 2;
+    if (coordinate < 0) {
+        coordinate = -coordinate - 1
     }
     var output = '';
     while (coordinate >= 0x20) {
         output += String.fromCharCode((0x20 | (coordinate & 0x1f)) + 63);
-        coordinate >>= 5;
+        coordinate /= 32;
     }
-    output += String.fromCharCode(coordinate + 63);
+    output += String.fromCharCode((coordinate | 0) + 63);
```

Comment:
This is intentionally a bitwise OR. In the `String.fromCharCode()` above the bitwise operation causes decimals to be discarded. I added the  `| 0` to remove decimals here too. I wasn't sure if a number with decimals would cause problems.  In my test it worked also without rounding, but you never know how different browsers treat it ...